### DISCUSSION TITLE
Access has been blocked by CORS policy

### DISCUSSION THREAD

**Discussion by zdoryk** [fastapi/fastapi#9836] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
app.add_middleware(
    CORSMiddleware,
    allow_origins=['*'],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)
```

### Description

I hosted FastAPI on Deta. Before deta migrate to new hosting all was fine, but after I've started to get this error with cors:

**Access to XMLHttpRequest at 'https://sfoodiedeta-1-w6589305.deta.app/wake-up' from origin 'https://sfoodie.live' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.**

### Operating System

Other

### Operating System Details

Actually idk which system Deta is using.

### FastAPI Version

0.78.0

### Python Version

3.9

### Additional Context

I am using Nuxt app that is hosted on the Vercel, when I am trying to reach local FastAPI app all is fine, but when try to reach deta app then it throw this error.

**Comment by csbasil** [comment]
Kindly try to use a list of origins in `allow_origins` instead of the wildcard

**Reply by zdoryk** [reply]
Like, list the origins that possibly can make requests to the endpoints?

**Reply by csbasil** [reply]
Yes. Like mentioned in the [docs](https://fastapi.tiangolo.com/tutorial/cors/).

**Reply by zdoryk** [reply]
I added oigins like this:


` origins = ["http://localhost:3000", "https://sfoodie.live", "https://sfoodie.vercel.app"]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)`



But still have the same problem:
Access to XMLHttpRequest at 'https://sfoodiedeta-1-w6589305.deta.app/wake-up' from origin 'https://sfoodie.live' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Reply by zdoryk** [reply]
UPD: I tried to do request via Postman and have this:
<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="UTF-8" />
	<link rel="icon" href="/auth_assets/favicon.png" />
	<meta name="viewport" content="width=device-width, initial-scale=1.0" />
	<title>Authorization | Deta Space</title>
	<script defer data-domain='deta.space' src='https://plausible.io/js/plausible.js'></script>
	<script type="module" crossorigin src="/auth_assets/js/index-02596b6e.js"></script>
	<link rel="stylesheet" href="/auth_assets/css/index-4a277ef8.css">
</head>

<body>
	<div id="app"></div>

</body>

</html>


So as you see Deta for some reason wants me to enter creds before using the API itself

**Reply by csbasil** [reply]
In that case, it may not be a FastAPI config issue. By the way I inspected https://sfoodie.live and not able see the CORS error.

**Reply by zdoryk** [reply]
Yes, soo... The problem was with Deta. When they migrated to new "Space" version they have this problem with CORS. I set visibility from private (was setted by default) to public and now it works. But they have a lot of issues that are related to CORS on their Discord channel.

**Comment by iudeen** [comment]
> If you want to allow credentials then your Access-Control-Allow-Origin must not use `*`

More info [here](https://stackoverflow.com/a/19744754).

**Comment by zdoryk** [comment, marked as answer]
Visibility on Deta need to be set on True:
v: 0
micros:
  - name: sfoodie-deta
    src: ./
    engine: python3.9
    public: true
    run: uvicorn main:app

**Reply by blogkhoahoc** [reply]
Have you thought of any way yet? I also have the same problem as you
